monai/bundle/config_parser.py

```diff
@@ -424,7 +424,20 @@ def load_config_files(cls, files: PathLike | Sequence[PathLike] | dict, **kwargs
             files = files.split(",")
         for i in ensure_tuple(files):
             for k, v in (cls.load_config_file(i, **kwargs)).items():
-                parser[k] = v
+                if k.startswith("+"):
```

Comment:
@borisfom Please add new unit tests in `test_config_parser.py` to cover this feature.
You also need to update this doc for syntax description: https://github.com/Project-MONAI/MONAI/blob/dev/docs/source/config_syntax.md.

@ericspod @KumoLiu What do you think about this new syntax?

Reply:
@ericspod @KumoLiu Here is an usage example: https://github.com/Project-MONAI/model-zoo/blob/a205bf469c2ee427b29dcbc9558712c690548cbd/models/pathology_nuclei_classification/configs/inference_trt.json

Reply:
@Nic-Ma : it may be worth noting that existing override-only handling of multiple config files neither described in config_syntax.md nor tested in test_config_parser.py.

Reply:
You are right.
1. `config_syntax.md` only describes syntax, I see you added new syntax `"+"` here, so request to add the doc.
2. We truly forget to add test cases for `load_config_files()`, could you please help add it with your new cases?

Thanks in advance.

Reply:
Hi @borisfom, thanks for adding this enhancement—it seems like a valuable addition. Previously, we override the arguments by setting the args_file input via monai.bundle.update_kwargs. However, it seems we may not have handled this well [here](https://github.com/Project-MONAI/MONAI/blob/1a8afd18d5fb132efa26d6f73c3bf1fdbccd985d/monai/bundle/scripts.py#L105).

And I have a quick try:
```
from monai.bundle.config_parser import ConfigParser
from monai.bundle.scripts import ConfigWorkflow, run

json1 = "/workspace/Code/MONAI/json-1.json"
json2 = "/workspace/Code/MONAI/json-2.json"

run("images", args_file=[json1, json2], config_file=json2)
```
The summary output will be:
```
2024-08-27 03:37:53,474 - INFO - --- input summary of monai.bundle.scripts.run ---
2024-08-27 03:37:53,474 - INFO - > +imports: ['$from monai.networks import trt_compile']
2024-08-27 03:37:53,475 - INFO - > network: '$trt_compile(@network_def.to(@device), @checkpoint)'
2024-08-27 03:37:53,475 - INFO - > imports: ['$import glob', '$import json', '$import pathlib', '$import os']
```
Will this change affect reading any properties since it will assume `+imports` as a new key.

Reply:
Done.

Reply:
@KumoLiu : Well it seems that load_config_files() does use different override rules compared to run(). 
Also, monai.bundle.update_kwargs only merges dictionaries, not lists. 
Perhaps, it would make sense to move the '+' merging code from load_config_files into an independent pass that would always be guaranteed to run last, before the first  use of config - and that would process all '+' entries. 
Merging is actually easier to accomplish universally, as this is default behavior if both json files would just be concatenated. That would be the most intuitive behavior. 
On the other hand - override behavior is also necessary, from what I understand.
I am not that knowledgeable about all possible control flows in MONAI, so if you can suggest a valid path to accomplish uniform behavior while allowing both override and merge, let's consider that.

Reply:
@KumoLiu : actually, I think we can unify the behavior by simply adding '+' prefix special case processing to update_kwargs... Hold on I will do that fix.

Reply:
Thanks for the quick update, looks much better now.
Hi @ericspod, do you have more concern based on this update?